libs/core/algorithms/include/hpx/parallel/unseq/simd_helpers.hpp

```diff
@@ -0,0 +1,161 @@
+//  Copyright (c) 2022 A Kishore Kumar
+//  Copyright (c) 2023 Hartmut Kaiser
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once
+
+#include <hpx/config.hpp>
+#include <hpx/functional/detail/invoke.hpp>
+#include <hpx/iterator_support/traits/is_iterator.hpp>
+#include <hpx/parallel/unseq/reduce_helpers.hpp>
+#include <hpx/type_support/construct_at.hpp>
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <type_traits>
+#include <utility>
+
+namespace hpx::parallel::util {
+    /*
+        Compiler and Hardware should also support vector operations for IterDiff,
+        else we see slower performance when compared to sequential version
+    */
+    template <class Iter, class IterDiff, class F>
+    Iter unseq_first_n(const Iter first, const IterDiff n, F&& f_) noexcept
+    {
+        /*
+            OMP loops can not have ++Iter, only integral types are allowed
+            Hence perform arthemetic on Iterators
+            which is O(1) only in case of random access iterators
+        */
+        static_assert(hpx::traits::is_random_access_iterator_v<Iter>,
+            "algorithm is efficient only in case of Random Access Iterator");
+#if HPX_EARLYEXIT_PRESENT
```

Comment:
Is this a build-system constant? Where would it be defined?

Reply:
If you are referring to `HPX_EARLYEXIT_PRESENT`, then its defined in [auto_vectorization.hpp](https://github.com/STEllAR-GROUP/hpx/pull/6286/files#diff-f9fca42172c7e4f2bed2865a1937603f7f3084c0dcff9af28e550dac67dfac2dR30).

It is a check for `#pragma omp simd early_exit` which is supported only by intel compilers newer than version 1800